### ISSUE TITLE
VHeaderFilter Typescript Conversion

### ISSUE THREAD

**Issue by ramadanomar** [WordPress/openverse-frontend#1753]
## Description
The types in VHeaderFilter aren't detected even though they are defined in ~/composables

## Screenshots

![image](https://user-images.githubusercontent.com/55840806/187225556-4d80d5ce-7692-45f9-be55-05dd37aa3a88.png)

## Additional context
Stumbled upon this while trying to figure out why tests results vary depending on the enviroment #1737. This error gets replaced by others if we include /src/components/VHeader/VHeaderFilter.vue to the tsconfig file.
![image](https://user-images.githubusercontent.com/55840806/187226668-2a85f3e7-8e92-46da-b89c-19f81626360b.png)

I suspect it is related to #1752. I am working on finding a fix. If you have any insight / ideas on how to approach this, i'd greatly appreciate it !

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

**Comment by obulat** [comment]
Have you run `pnpm install` after checking out the PR (or are you running into these errors on main), @ramadanomar? The version of `vue-tsc` has been updated recently, and different versions produce different type errors. 
Also, there are currently several PRs that update the types and modals, it might be better to wait until they land to try to figure out what causes this issue.

**Comment by ramadanomar** [comment]
Okay! I'll wait for the new PRs to be merged until continuing with this issue.

**Comment by obulat** [comment]
I tried looking into it, and it appears to be a VSCode/Volar problem. I don't see these problems in WebStorm, and you don't get errors when you run `pnpm types`.

The problem seems to be that VSCode or Volar are not picking up the `paths` object in the `tsconfig` file, and marking all the aliased paths that start with `~` as not found. If you replace all of the aliased paths with the relative paths ( `'~/utils/focus-management' with ` `'../../utils/focus-management'`), the problem disappears.

**Comment by ramadanomar** [comment]
> The problem seems to be that VSCode or Volar are not picking up the `paths` object in the `tsconfig` file

I think it's because we don't include it in tsconfig.json 

```json
{ /* ... */
"include": [
  "**/*.ts",
/*rest of the files */
  "src/components/VHeader/VHeaderFilter.vue",
 ],
 }
```

Adding something like that to `tsconfig.json` fixes the problem but we get some type errors "Type HTMLElement | null " is not assignable. Weird thing is it doesn't fail the pnpm test linting check.

**Comment by ramadanomar** [comment]
Nvm it fails when i try to upload a branch.

**Comment by obulat** [comment]
@ramadanomar, I've added `lang="ts"` to the VHeader components to be able to use TypeScript when writing them. I did not add them to `tsconfig` so that they wouldn't raise TypeScript errors (see more details in [this comment](https://github.com/WordPress/openverse-frontend/pull/1472#issuecomment-1155391012)). Normally, this can raise warnings in the IDE, but should not cause errors.

We have vendored-in the Reakit functions, and I plan to convert VHeader, VModal and VPopover components to valid Typescript and add them to TSConfig after the modal refactor PRs are merged in.
